src/features/move/move.ts

```diff
@@ -328,42 +317,64 @@ export class MorphEdgesAnimation extends Animation {
     }
 
     start() {
-        this.expandedMementi.forEach(memento => {
-            memento.edge.removeAll(e => e instanceof SRoutingHandle);
+        this.expanded.forEach(morph => {
+            morph.memento.edge.removeAll(e => e instanceof SRoutingHandle);
         });
         return super.start();
     }
 
     tween(t: number) {
         if (t === 1) {
-            this.originalMementi.forEach(memento => {
+            this.expanded.forEach(morph => {
+                const memento = morph.memento;
                 if (this.reverse)
-                    memento.after.router.applySnapshot(memento.edge, memento.before);
+                    memento.before.router.applySnapshot(memento.edge, memento.before);
                 else
                     memento.after.router.applySnapshot(memento.edge, memento.after);
             });
         } else {
-            this.expandedMementi.forEach(memento => {
+            this.expanded.forEach(morph => {
                 const newRoutingPoints: Point[] = [];
-                for (let i = 0; i < memento.before.routingPoints.length; ++i) {
-                    const startPoint = memento.before.routingPoints[i];
-                    const endPoint = memento.after.routingPoints[i];
-                    newRoutingPoints.push({
-                        x: (1 - t) * startPoint.x + t * endPoint.x,
-                        y: (1 - t) * startPoint.y + t * endPoint.y
-                    });
-                }
-                const closestSnapshot = t < 0.5 ? memento.before : memento.after;
+                // ignore source and target anchor
+                for (let i = 1; i < morph.startExpandedRoute.length - 1; ++i)
+                    newRoutingPoints.push(linear(morph.startExpandedRoute[i], morph.endExpandedRoute[i], t));
+
+                const closestSnapshot = t < 0.5 ? morph.memento.before : morph.memento.after;
                 const newSnapshot: EdgeSnapshot = {
                     ...closestSnapshot,
                     routingPoints: newRoutingPoints,
                     routingHandles: []
                 };
-                closestSnapshot.router.applySnapshot(memento.edge, newSnapshot);
+                closestSnapshot.router.applySnapshot(morph.memento.edge, newSnapshot);
             });
         }
         return this.model;
     }
+
+    protected growToSize(route: RoutedPoint[], targetSize: number): Point[] {
```

Comment:
This is very similar to `EdgeMorphAnimation.interpolateMissingPoints`. Could that be reused or extracted?

Reply:
`EdgeMorphAnimation` has been deleted